datacube/api/core.py

```diff
@@ -256,6 +256,7 @@ def load(self,
              progress_cbk: ProgressFunction | None = None,
              patch_url: Callable[[str], str] | None = None,
              limit: int | None = None,
+             driver: Any | None = None,
```

Comment:
New dc.load argument should be documented in doc string.  Can be a simple placeholder for now, but would be good to have something.